astropy/io/fits/diff.py

```diff
@@ -705,30 +705,30 @@ def get_header_values_comments(cards):
     def _report(self):
         if self.diff_keyword_count:
             self._writeln(u(' Headers have different number of cards:'))
-            self._writeln(u('  a: %d') % self.diff_keyword_count[0])
-            self._writeln(u('  b: %d') % self.diff_keyword_count[1])
+            self._writeln(u('  a: {}').format(self.diff_keyword_count[0]))
+            self._writeln(u('  b: {}').format(self.diff_keyword_count[1]))
         if self.diff_keywords:
             for keyword in self.diff_keywords[0]:
                 if keyword in Card._commentary_keywords:
                     val = self.a[keyword][0]
                 else:
                     val = self.a[keyword]
-                self._writeln(u(' Extra keyword %-8r in a: %r') %
-                              (keyword, val))
+                self._writeln(u(' Extra keyword {!r} in a: {!r}').format(
+                                keyword, val))
             for keyword in self.diff_keywords[1]:
                 if keyword in Card._commentary_keywords:
                     val = self.b[keyword][0]
                 else:
                     val = self.b[keyword]
-                self._writeln(u(' Extra keyword %-8r in b: %r') %
-                              (keyword, val))
+                self._writeln(u(' Extra keyword {!r} in b: {!r}').format(
+                                keyword, val))
 
         if self.diff_duplicate_keywords:
             for keyword, count in sorted(self.diff_duplicate_keywords.items()):
-                self._writeln(u(' Inconsistent duplicates of keyword %-8r:') %
-                              keyword)
-                self._writeln(u('  Occurs %d time(s) in a, %d times in (b)') %
-                              count)
+                self._writeln(u(' Inconsistent duplicates of keyword '
+                                '{!r}:').format(keyword))
```

Comment:
Tefault alignment differs between old and new style, ``-`` means left alignment and we get it as def. I got all weird failures for leaving the 8 with the ``r`` formatting, so took my chances and removed it altogether. It made the tests happy... 
(basically the fits and votable modules were the ones with non obvious cases.)